pool/pool.go

```diff
@@ -33,22 +33,373 @@ import (
 
 // client is a wrapper for sdkClient.Client to generate mock.
 type client interface {
-	BalanceGet(context.Context, sdkClient.PrmBalanceGet) (*sdkClient.ResBalanceGet, error)
-	ContainerPut(context.Context, sdkClient.PrmContainerPut) (*sdkClient.ResContainerPut, error)
-	ContainerGet(context.Context, sdkClient.PrmContainerGet) (*sdkClient.ResContainerGet, error)
-	ContainerList(context.Context, sdkClient.PrmContainerList) (*sdkClient.ResContainerList, error)
-	ContainerDelete(context.Context, sdkClient.PrmContainerDelete) (*sdkClient.ResContainerDelete, error)
-	ContainerEACL(context.Context, sdkClient.PrmContainerEACL) (*sdkClient.ResContainerEACL, error)
-	ContainerSetEACL(context.Context, sdkClient.PrmContainerSetEACL) (*sdkClient.ResContainerSetEACL, error)
-	EndpointInfo(context.Context, sdkClient.PrmEndpointInfo) (*sdkClient.ResEndpointInfo, error)
-	NetworkInfo(context.Context, sdkClient.PrmNetworkInfo) (*sdkClient.ResNetworkInfo, error)
-	ObjectPutInit(context.Context, sdkClient.PrmObjectPutInit) (*sdkClient.ObjectWriter, error)
-	ObjectDelete(context.Context, sdkClient.PrmObjectDelete) (*sdkClient.ResObjectDelete, error)
-	ObjectGetInit(context.Context, sdkClient.PrmObjectGet) (*sdkClient.ObjectReader, error)
-	ObjectHead(context.Context, sdkClient.PrmObjectHead) (*sdkClient.ResObjectHead, error)
-	ObjectRangeInit(context.Context, sdkClient.PrmObjectRange) (*sdkClient.ObjectRangeReader, error)
-	ObjectSearchInit(context.Context, sdkClient.PrmObjectSearch) (*sdkClient.ObjectListReader, error)
-	SessionCreate(context.Context, sdkClient.PrmSessionCreate) (*sdkClient.ResSessionCreate, error)
+	BalanceGet(context.Context, PrmBalanceGet) (*accounting.Decimal, error)
+	ContainerPut(context.Context, PrmContainerPut) (*cid.ID, error)
+	ContainerGet(context.Context, PrmContainerGet) (*container.Container, error)
+	ContainerList(context.Context, PrmContainerList) ([]cid.ID, error)
+	ContainerDelete(context.Context, PrmContainerDelete) error
+	ContainerEACL(context.Context, PrmContainerEACL) (*eacl.Table, error)
+	ContainerSetEACL(context.Context, PrmContainerSetEACL) error
+	EndpointInfo(context.Context, PrmEndpointInfo) (*netmap.NodeInfo, error)
+	NetworkInfo(context.Context, PrmNetworkInfo) (*netmap.NetworkInfo, error)
+	ObjectPut(context.Context, PrmObjectPut) (*oid.ID, error)
+	ObjectDelete(context.Context, PrmObjectDelete) error
+	ObjectGet(context.Context, PrmObjectGet) (*ResGetObject, error)
+	ObjectHead(context.Context, PrmObjectHead) (*object.Object, error)
+	ObjectRange(context.Context, PrmObjectRange) (*ResObjectRange, error)
+	ObjectSearch(context.Context, PrmObjectSearch) (*ResObjectSearch, error)
+	SessionCreate(context.Context, PrmCreateSession) (*ResCreateSession, error)
+}
+
+type clientWrapper struct {
+	client *sdkClient.Client
```

Comment:
What about  the following warning when we creates new wrapper using value rather than pointer (`return &clientWrapper{client: c, key: prm.key}, c.Dial(prmDial)`):
```
Literal copies a lock value from 'c': type 'sdkClient.Client' contains 'client.Client' contains 'sync.Once' contains 'sync.Mutex' which is 'sync.Locker' 
```

Reply:
Of course we need to change the function code:
```
	var prmInit sdkClient.PrmInit
	prmInit.ResolveNeoFSFailures()
	prmInit.SetDefaultPrivateKey(prm.key)

	res := &clientWrapper{key: prm.key}

	res.client.Init(prmInit)

	var prmDial sdkClient.PrmDial
	prmDial.SetServerURI(prm.address)
	prmDial.SetTimeout(prm.timeout)

	err := c.Dial(prmDial)
	if err != nil {
		return nil, fmt.Errorf("client dial: %w", err)
	}

	return res, nil
```